commands/cli_nukeallresources_test.go

```diff
@@ -0,0 +1,547 @@
+	// Generate test bucket and test instance name
+	bucketName := aws.S3GenBucketName()
+	ec2InstanceName := "cloud-nuke-test-" + util.UniqueID()
+
+	resourceTypeIdentifierMap := map[string]string{}
+	resourceTypes := []string{}
+	tags := []map[string]string{{"Key": "cloud-nuke-test", "Value": "true"}}
+
+	// Create test bucket and test instance
+	resourceTypeIdentifierMap, err := createTestResources(
+		awsParams.AWSSession,
+		[]createTestResourcesArgs{
+			{resourceType: "s3", name: bucketName, tags: tags},
+			{resourceType: "ec2", name: ec2InstanceName, tags: tags},
+		},
+	)
+
+	// Ensure test bucket + test EC2 instance gets deleted at the end of the test
+	// Add defer before checking error to handle the case where partial resources get
+	// created and must be destroyed
+	defer aws.NukeAllResources(buildNukeAllResourcesArgs(awsParams.AWSSession, resourceTypeIdentifierMap, awsParams.Region, args.ignoreErrResourceTypes))
+
+	require.NoError(t, err, "Failed to create test resources")
+
+	for resourceType := range resourceTypeIdentifierMap {
+		resourceTypes = append(resourceTypes, resourceType)
+	}
+
+	// Assume role
+	var roleSession *session.Session
+	if assumeRoleArn == "" {
+		roleSession = awsParams.AWSSession
+	} else {
+		roleSession, err = aws.AssumeIAMRole(assumeRoleArn, awsParams.Region)
+		require.NoErrorf(t, err, "Failed to assume role - %s - %s", assumeRoleArn, err)
+	}
+
+	excludeAfter, err := parseDurationParam("0s")
+	require.NoError(t, err, "parseDurationParam failed")
+
+	// Do assumed role get for existing resources of type EC2 and S3
+	account, err := aws.GetAllResources(
+		aws.GetAllResourcesArgs{
+			TargetRegions:          []string{awsParams.Region},
+			ExcludeAfter:           *excludeAfter,
+			NukeResourceTypes:      resourceTypes,
+			IgnoreErrResourceTypes: args.ignoreErrResourceTypes,
+			RegionSessionMap:       map[string]*session.Session{awsParams.Region: roleSession},
+		},
+	)
+	// Validate that get call should fail if specified
+	if args.getResourcesShouldFail {
+		require.Error(t, err, "GetAllResources did not fail")
+		logging.Logger.Debug("Expected: GetAllResources failed")
+		return
+	}
+	require.NoError(t, err, "GetAllResources failed")
+	logging.Logger.Debug("Expected: GetAllResources passed")
+
+	// Validate that get call should return the right resource types
+	if collections.ListContains(args.getResourcesOutputTypes, "ec2") {
+		require.Truef(
+			t, resourceExists(awsParams.Region, "ec2", resourceTypeIdentifierMap["ec2"], account),
+			"Failed to find test EC2 instance - name - %s - instanceID - %s", ec2InstanceName, resourceTypeIdentifierMap["ec2"],
+		)
+		logging.Logger.Debug("Expected: Post get check: ec2 exists")
+	}
+	if collections.ListContains(args.getResourcesOutputTypes, "s3") {
+		require.Truef(
+			t, resourceExists(awsParams.Region, "s3", bucketName, account),
+			"Failed to find test S3 bucket - %s", bucketName,
+		)
+		logging.Logger.Debug("Expected: Post get check: s3 exists")
+	}
```

Comment:
Seems like you could loop over `args.getResourceOutputTypes` and do this check dynamically, rather than hard-coding ec2 and s3 checks?